{"version":3,"file":"consoleUtils.js","names":["window","console","log","warn","error","_origAssert","assert","conditional","msg","call","Error"],"sources":["../../../../../static/rb/js/utils/consoleUtils.es6.js"],"sourcesContent":["(function() {\n\n\nif (typeof window.console === 'undefined') {\n    window.console = {};\n}\n\nif (typeof console.log === 'undefined') {\n    console.log = function() {};\n}\n\nif (typeof console.warn === 'undefined') {\n    console.warn = function() {};\n}\n\nif (typeof console.error === 'undefined') {\n    console.error = function() {};\n}\n\n\nconst _origAssert = console.assert;\n\n/*\n * console.assert may not behave as we'd hope on all implementations.\n * On Chrome, for instance, it doesn't raise an exception. So, fall back\n * to raising one.\n */\nconsole.assert = function(conditional, msg) {\n    if (_origAssert && _origAssert.call) {\n        _origAssert.call(console, conditional, msg);\n    }\n\n    /* If the above assert never raised an exception, raise our own. */\n    if (!conditional) {\n        throw new Error(msg);\n    }\n};\n\n\n})();\n"],"mappings":";;AAAA,CAAC,YAAW;EAGZ,IAAI,OAAOA,MAAM,CAACC,OAAd,KAA0B,WAA9B,EAA2C;IACvCD,MAAM,CAACC,OAAP,GAAiB,EAAjB;EACH;;EAED,IAAI,OAAOA,OAAO,CAACC,GAAf,KAAuB,WAA3B,EAAwC;IACpCD,OAAO,CAACC,GAAR,GAAc,YAAW,CAAE,CAA3B;EACH;;EAED,IAAI,OAAOD,OAAO,CAACE,IAAf,KAAwB,WAA5B,EAAyC;IACrCF,OAAO,CAACE,IAAR,GAAe,YAAW,CAAE,CAA5B;EACH;;EAED,IAAI,OAAOF,OAAO,CAACG,KAAf,KAAyB,WAA7B,EAA0C;IACtCH,OAAO,CAACG,KAAR,GAAgB,YAAW,CAAE,CAA7B;EACH;;EAGD,MAAMC,WAAW,GAAGJ,OAAO,CAACK,MAA5B;EAEA;AACA;AACA;AACA;AACA;;EACAL,OAAO,CAACK,MAAR,GAAiB,UAASC,WAAT,EAAsBC,GAAtB,EAA2B;IACxC,IAAIH,WAAW,IAAIA,WAAW,CAACI,IAA/B,EAAqC;MACjCJ,WAAW,CAACI,IAAZ,CAAiBR,OAAjB,EAA0BM,WAA1B,EAAuCC,GAAvC;IACH;IAED;;;IACA,IAAI,CAACD,WAAL,EAAkB;MACd,MAAM,IAAIG,KAAJ,CAAUF,GAAV,CAAN;IACH;EACJ,CATD;AAYC,CAvCD"}